cmake/common.cmake

```diff
@@ -189,70 +199,206 @@ Unit Test Specification
                 file(WRITE ${_unit_test_results_rst} "
 Unit Test Results
 =================
 
 .. test-report:: Unit Test Results
-    :id: TEST_RESULT
-    :file: {{ component_test_junit_xml }}
+    :id: TEST_RESULT_${component_name}
+    :file: ${_component_test_junit_xml}
+
+")
+
+                # create the code coverage rst file
+                set(_coverage_rst ${_component_reports_out_dir}/coverage.rst)
+                file(WRITE ${_coverage_rst} "
+Code Coverage
+=============
+
+`Report <coverage/index.html>`_
 
 ")
-                set(_component_doxyfile ${SPHINX_OUTPUT_DIR}/Doxyfile)
 
                 # generate Doxyfile from template
-                set(DOXYGEN_PROJECT_NAME "Doxygen Documentation")
-                set(DOXYGEN_OUTPUT_DIRECTORY ${SPHINX_OUTPUT_DIR}/doxygen)
+                set(_component_doxyfile ${_component_reports_out_dir}/Doxyfile)
+                set(DOXYGEN_PROJECT_NAME "${CREATE_COMPONENT_LONG_NAME} Documentation")
+                set(DOXYGEN_OUTPUT_DIRECTORY ${_component_reports_out_dir}/doxygen)
                 set(DOXYGEN_INPUT "${_component_dir}/src ${_component_dir}/test ${KCONFIG_OUT_DIR}")
 
                 # We need to add the googletest include directory to the doxygen include path
                 # to be able to resolve the TEST() macros in the test files.
-                set(DOXYGEN_INCLUDE_PATH "${SPHINX_SOURCE_DIR}/build/modules/googletest-src/googletest/include ${KCONFIG_OUT_DIR}")
-                set(DOXYGEN_AWESOME_PATH "${SPHINX_SOURCE_DIR}/doc/doxygen-awesome")
-                configure_file(${SPHINX_SOURCE_DIR}/doc/Doxyfile.in ${_component_doxyfile} @ONLY)
+                set(DOXYGEN_INCLUDE_PATH "${_sphinx_source_dir}/build/modules/googletest-src/googletest/include ${KCONFIG_OUT_DIR}")
+                set(DOXYGEN_AWESOME_PATH "${_sphinx_source_dir}/doc/doxygen-awesome")
+                configure_file(${_sphinx_source_dir}/doc/Doxyfile.in ${_component_doxyfile} @ONLY)
                 file(RELATIVE_PATH _rel_component_doxyfile ${CMAKE_CURRENT_BINARY_DIR} ${_component_doxyfile})
-                file(RELATIVE_PATH _rel_component_doxysphinx_index_rst ${SPHINX_SOURCE_DIR} ${DOXYGEN_OUTPUT_DIRECTORY}/html/index)
+                file(RELATIVE_PATH _rel_component_doxysphinx_index_rst ${_sphinx_source_dir} ${DOXYGEN_OUTPUT_DIRECTORY}/html/index)
 
                 file(WRITE ${_reports_config_json} "{
-                    \"generated_rst_content\": \".. toctree::\\n    :maxdepth: 2\\n\\n    /${_rel_component_doc_dir}/index\\n    /${_rel_unit_test_spec_rst}\\n    /${_rel_unit_test_results_rst}\\n    /${_rel_component_doxysphinx_index_rst}\",
-                    \"component_doc_dir\": \"${_rel_component_doc_dir}\", 
-                    \"component_test_junit_xml\": \"${_component_test_junit_xml}\",
-                    \"include_patterns\": [\"${_rel_component_doc_dir}/**\",\"${_rel_sphinx_output_dir}/**\"]
-                }")
+    \"component_info\": ${_component_info},

```

Comment:
No need to store multiple keys for a component. All component relevant info is stored in component_info